sunpy/map/mapbase.py

```diff
@@ -2052,7 +2052,8 @@ def peek(self, draw_limb=False, draw_grid=False,
                 pad = 0.12  # Pad to compensate for ticks and axes labels
             else:
                 pad = 0.05  # Default value for vertical colorbar
-            figure.colorbar(im, pad=pad)
+            figure.colorbar(im, pad=pad).set_label(str(self.unit)
+                                                    if self.unit is not None else "", rotation=0, labelpad = -50, y=-0.02, size=12)
```

Comment:
I think doing the "str(self.unit)" section on a new line and passing the result into color bar would be neater.